### PR TITLE
backupccl: handle RESTORE of TTL tables

### PR DESCRIPTION
**Pull request by otan**
This commit adds schedules to restored table descriptors.

Release note: None

### CODE REVIEW

File: pkg/ccl/backupccl/restore_job.go [hunk 1]
```diff
@@ -1072,6 +1072,49 @@ func createImportingDescriptors(
...
+							shouldCreateScheduledJob = true
+						} else {
+							var args catpb.ScheduledRowLevelTTLArgs
+							if err := pbtypes.UnmarshalAny(existingJob.ExecutionArgs().Args, &args); err != nil || args.TableID != table.GetID() {
+								shouldCreateScheduledJob = true
+							}
+						}
+
+						if shouldCreateScheduledJob {
+							j, err := sql.CreateRowLevelTTLScheduledJob(
```
Comment (stevendanna) [inline pkg/ccl/backupccl/restore_job.go:1101]: Do we need to clean these up in OnFailOrCancel? I see that these jobs are automatically dropped in the schema changer when we process a dropped table descriptor, but our dropDescriptor method seems to do a lot of the heavy lifting itself so it isn't clear to me that the logic in the schema changer will be in play here.  

If this is automatically handled and I've just misread things, perhaps we could add a test to ensure they are cleaned up?

**Review by stevendanna — CHANGES_REQUESTED** [review]

File: pkg/ccl/backupccl/restore_job.go (hunk 1 above)
Comment (stevendanna) [reply pkg/ccl/backupccl/restore_job.go:1101]: `TestRestoreFailCleansUpTypeBackReferences` or `TestRestoreFailDatabaseCleanup` look like they might be useful to start from.

Comment (otan) [reply pkg/ccl/backupccl/restore_job.go]: done!

**Comment by otan** [comment]
@stevendanna do you mind giving this another squiz?

**Review by stevendanna — APPROVED** [review]
This looks reasonable to me.  I believe you originally had some code for checking for existing schedules to see if one applied to this table, but I assume that was removed since we don't allow restoring tables/databases on top of existing table/databases so the existing schedules can be assumed to not exist.

**Comment by otan** [comment]
yep, that and i reset the scheduleid to 0 when taking it in for the first time.

thanks!

bors r=stevendanna

**Pull request merged**